tendermint/src/abci/kind.rs

```diff
@@ -0,0 +1,23 @@
+/// A category of ABCI method.
+///
+/// ABCI methods are split into four categories. Tendermint opens one ABCI
+/// connection for each category and refers to these categories as *connections*,
+/// but nothing actually restricts an ABCI connection from calling methods in
+/// multiple categories.
+///
+/// This enum breaks out the `Flush` method as a distinct category, since it is
+/// used to control the execution of other methods.
+pub enum MethodKind {
```

Comment:
While I know it's correct to have some kind of indication as to which requests apply to which connection from a modelling perspective, is it actually useful to an ABCI developer? Do you have any particular use cases in mind?

Reply:
Yep, in https://github.com/hdevalence/tower-abci I use this to split different requests into different queues that can either go to different backing services (e.g., info requests are processed by a replica of the state), or to the same service with different priorities, etc.